app/assets/javascripts/camaleon_cms/admin/_libraries.js

```diff
@@ -1,136 +1,168 @@
+// eslint-disable-next-line no-unused-expressions, no-void, no-sequences, no-useless-escape, no-mixed-operators, no-return-assign
+!(function(a) { typeof define === 'function' && define.amd ? define(['jquery'], function(b) { a(b) }) : typeof module === 'object' && typeof module.exports === 'object' ? module.exports = a(require('jquery')) : a(window.jQuery) }(function(a) { 'use strict'; function b(a) { void 0 === a && (a = window.navigator.userAgent), a = a.toLowerCase(); const b = /(edge)\/([\w.]+)/.exec(a) || /(opr)[\/]([\w.]+)/.exec(a) || /(chrome)[ \/]([\w.]+)/.exec(a) || /(version)(applewebkit)[ \/]([\w.]+).*(safari)[ \/]([\w.]+)/.exec(a) || /(webkit)[ \/]([\w.]+).*(version)[ \/]([\w.]+).*(safari)[ \/]([\w.]+)/.exec(a) || /(webkit)[ \/]([\w.]+)/.exec(a) || /(opera)(?:.*version|)[ \/]([\w.]+)/.exec(a) || /(msie) ([\w.]+)/.exec(a) || a.indexOf('trident') >= 0 && /(rv)(?::| )([\w.]+)/.exec(a) || a.indexOf('compatible') < 0 && /(mozilla)(?:.*? rv:([\w.]+)|)/.exec(a) || []; const c = /(ipad)/.exec(a) || /(ipod)/.exec(a) || /(iphone)/.exec(a) || /(kindle)/.exec(a) || /(silk)/.exec(a) || /(android)/.exec(a) || /(windows phone)/.exec(a) || /(win)/.exec(a) || /(mac)/.exec(a) || /(linux)/.exec(a) || /(cros)/.exec(a) || /(playbook)/.exec(a) || /(bb)/.exec(a) || /(blackberry)/.exec(a) || []; const d = {}; const e = { browser: b[5] || b[3] || b[1] || '', version: b[2] || b[4] || '0', versionNumber: b[4] || b[2] || '0', platform: c[0] || '' }; if (e.browser && (d[e.browser] = !0, d.version = e.version, d.versionNumber = parseInt(e.versionNumber, 10)), e.platform && (d[e.platform] = !0), (d.android || d.bb || d.blackberry || d.ipad || d.iphone || d.ipod || d.kindle || d.playbook || d.silk || d['windows phone']) && (d.mobile = !0), (d.cros || d.mac || d.linux || d.win) && (d.desktop = !0), (d.chrome || d.opr || d.safari) && (d.webkit = !0), d.rv || d.edge) { const f = 'msie'; e.browser = f, d[f] = !0 } if (d.safari && d.blackberry) { const g = 'blackberry'; e.browser = g, d[g] = !0 } if (d.safari && d.playbook) { const h = 'playbook'; e.browser = h, d[h] = !0 } if (d.bb) { const i = 'blackberry'; e.browser = i, d[i] = !0 } if (d.opr) { const j = 'opera'; e.browser = j, d[j] = !0 } if (d.safari && d.android) { const k = 'android'; e.browser = k, d[k] = !0 } if (d.safari && d.kindle) { const l = 'kindle'; e.browser = l, d[l] = !0 } if (d.safari && d.silk) { const m = 'silk'; e.browser = m, d[m] = !0 } return d.name = e.browser, d.platform = e.platform, d } return window.jQBrowser = b(window.navigator.userAgent), window.jQBrowser.uaMatch = b, a && (a.browser = window.jQBrowser), window.jQBrowser }))
```

Comment:
I think this abandoned `jquery-browser-plugin` code should be replaced eith with `bowser` or `modernizr` package, especially that the latter is actually in use by our app.

Also, is it just me or there is an error in this code: `void 0 === a && (a = window.navigator.userAgent), a = a.toLowerCase();` - seems wrong to me.

Reply:
I didn't think modernizr was relevant to modern browsers. I thought we'd probably stop using it, rather than use it more. Do we even need feature detection anymore?

As far as that code, I don't understand is how the comma acts in the expression. That's a JS pattern I haven't worked with before.

Reply:
RubyMine is also showing the error on the comma.

Also, I'm not sure if we are using browser feature detection. Maybe @owen2345 can enlighten us on these questions.